src/main/java/nextstep/subway/domain/LineStations.java

```diff
@@ -0,0 +1,111 @@
+package nextstep.subway.domain;
+
+import nextstep.subway.exception.ElementNotFoundException;
+import nextstep.subway.exception.InvalidParameterException;
+
+import javax.persistence.Embeddable;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.List;
+
+@Embeddable
+public class LineStations {
+
+    @OneToMany(fetch = FetchType.LAZY)
+    @JoinColumn(name = "line_id")
+    private List<LineStation> lineStations = new ArrayList<>();
+
+    public void add(LineStation lineStation) {
+        this.lineStations.add(lineStation);
+    }
+
+    public void infixSection(LineStation infixLineStation) {
+        LineStation needChangedStation = getNeedChangedStationStation(infixLineStation);
+
+        if (needChangedStation == null) {
+            add(infixLineStation);
+            return;
+        }
+        checkValidationParameter(needChangedStation, infixLineStation);
+
+        if (needChangedStation.isEqualsId(infixLineStation.getStation().getId())) {
+            infixLineStation.resetStation(infixLineStation.getPreStation());
+            infixLineStation.resetPreStation(needChangedStation.getPreStation());
+        }
```

Comment:
`기존 구간 사이에 새로운 역을 등록하는 경우`이고 `기존 구간의 하행선과 등록하려는 구간의 하행선이 일치하는 경우`에
`기존구간의 하행선`은 `등록하려는 구간의 상행선`으로 변경되고 `기존구간의 거리`는 `등록하려는 구간의 거리`만큼 감소되어야겠죠.
```suggestion
        if (needChangedStation.isEqualsId(infixLineStation.getStation().getId())) {
            needChangedStation.resetStation(infixLineStation.getPreStation());
            needChangedStation.거리감소(infixLineStation.getDistance());
        }
```
이렇게 되어야하지 않을까 싶네요! 😀

그리고 이렇게 하는것보다
`needChangedStation.fix(infixLineStation);`처럼 메시지를 던지면 적절히 역할을 위임할수 있을것같은데 어떻게 생각하시나요?

Reply:
네 맞아요! 저 if문 안에서 변경이 필요한 station에서 station 정보 + 거리정보를 모두 변경해야 하니,
각각의 메소드를 호출하는거보다 하나의 메소드로 메세지를 던져서 변경이 필요한 모든 정보를 한번에 변경하는 것이 좋을 것 같아요!